test/StrategyManager/createStrategy.test.ts

```diff
@@ -107,24 +107,25 @@ describe('StrategyManager#createStrategy', () => {
     })
 
     describe('REVERTS', () => {
-        it('if msg.sender doenst have an active subscription', async () => {
+        it('if msg.sender doesn\'t have an active subscription', async () => {
             const tx = strategyManager.connect(account1).createStrategy({
                 dcaInvestments: dcaStrategyPositions,
                 vaultInvestments: vaultStrategyPosition,
                 liquidityInvestments: [],
                 tokenInvestments: [],
                 permit: await subscriptionSignature.signSubscriptionPermit(
                     await account1.getAddress(),
-                    deadline,
+                    0,
                 ),
                 metadataHash: nameBioHash,
             })
 
-            expect(tx).to.be.revertedWithCustomError(strategyManager, 'Unauthorized')
+            await expect(tx).to.be.revertedWithCustomError(strategyManager, 'Unauthorized')
         })
 
         it('if strategy uses more than 20 dca investments', async () => {
             const investments: InvestLib.DcaInvestmentStruct[] = new Array(21)
+                .fill(0)
```

Comment:
I find it more straightforward to use the `Array.from()` method :
```ts
Array.from({ length: 21 }, () => ({ percentage: 0, poolId: 0, swaps: 10 }))
```
Its the same as `Array.from(obj).map(mapFn)`, except that it doesn't create an intermediary array.